src/Backend/Modules/MediaLibrary/Ajax/MoveMediaFolder.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Ajax;
+
+use Backend\Core\Engine\Base\AjaxAction as BackendBaseAJAXAction;
+use Backend\Core\Language\Language;
+use Backend\Modules\MediaLibrary\Domain\MediaFolder\Command\UpdateMediaFolder;
+use Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolder;
+use Backend\Modules\MediaLibrary\Domain\MediaFolder\Event\MediaFolderUpdated;
+
+/**
+ * This edit-action will reorder moved pages using Ajax
+ */
+class MoveMediaFolder extends BackendBaseAJAXAction
+{
+    /**
+     * Execute the action
+     */
+    public function execute()
+    {
+        // call parent
+        parent::execute();
+
+        /** @var MediaFolder $mediaFolder */
+        $mediaFolder = $this->getMediaFolder();
+
+        /** @var UpdateMediaFolder $updateMediaFolder */
+        $updateMediaFolder = new UpdateMediaFolder($mediaFolder);
+
+        /** @var MediaFolder|null $mediaFolder */
+        $droppedOnMediaFolder = $this->getMediaFolderWhereDroppedOn();
```

Comment:
I would extract this part (till line 48) into a separate method so you can use early returns instead of this if else nesting